scipy/stats/tests/test_stats.py

```diff
@@ -4178,6 +4178,15 @@ def test_nan_policy(self):
         assert_almost_equal(stats.kruskal(x, x, nan_policy='omit'), (0.0, 1.0))
         assert_raises(ValueError, stats.kruskal, x, x, nan_policy='raise')
         assert_raises(ValueError, stats.kruskal, x, x, nan_policy='foobar')
+        
+    def test_large_no_samples(self):
+        # Test to see if large samples are handled correctly.
+        n = 50000
+        x = np.random.randn(n)
+        y = np.random.randn(n) + 50
+        h, p = stats.kruskal(x, y)
+        expected = 0
+        assert_approx_equal(p, expected)
```

Comment:
This test passes now as well, looks like you want to test the value of ``h`` as well:

    In [17]: n = 5000

    In [18]: y = np.random.randn(n)

    In [19]: x = np.random.randn(n)

    In [20]: stats.kruskal(x, y)
    Out[20]: KruskalResult(statistic=0.11486519704703824, pvalue=0.73467174156664328)

    In [21]: n = 50000

    In [22]: y = np.random.randn(n)

    In [23]: x = np.random.randn(n)

    In [24]: stats.kruskal(x, y)
    Out[24]: KruskalResult(statistic=1827449.5560047459, pvalue=0.0)